pydantic/schema.py

```diff
@@ -726,6 +740,8 @@ def field_singleton_schema(  # noqa: C901 (ignore complexity)
         else:
             nested_models.add(model_name)
         schema_ref = {'$ref': ref_prefix + model_name}
```

Comment:
https://github.com/samuelcolvin/pydantic/pull/1611/files#r446530167
```suggestion
        schema_ref: Dict[str, Any] = {'$ref': ref_prefix + model_name}
```